### ISSUE TITLE
Filters.eq Doesn't Work With String Objects

### ISSUE THREAD

**Issue by gursahibsahni** [MorphiaOrg/morphia#2853]
**Describe the bug**
Using Filters.eq for String based fields doesn't work. Example:

`// DOES NOT work`
`datastore.find(CustomerDAO.class).filter(Filters.eq("activePhone", activePhone)).iterator().tryNext();`

`// DOES work`
`datastore.find(CustomerDAO.class).filter(Filters.eq("_id", new ObjectId(id))).iterator().tryNext();`

`// mongo client DOES work`
`datastore.getDatabase().getCollection("customer").find(eq("activePhone", activePhone));`

**Expected behavior**
Morphia should return data based on the activePhone data
![image](https://github.com/MorphiaOrg/morphia/assets/36925400/1e6edb1c-3915-4637-a6aa-d28a11d32b3c)

** Please complete the following information: **
 - Mongo Version - 4.11.1
 - Morphia Version: 2.4.11

**Comment by evanchooly** [comment]
Any particular reason you're querying against `CustomerDAO` and not `Customer`?

**Comment by gursahibsahni** [comment]
> Any particular reason you're querying against `CustomerDAO` and not `Customer`?

Doesn't matter -- CustomerDAO's collection name is `customer`. They both point to the same thing. 

<img width="401" alt="image" src="https://github.com/MorphiaOrg/morphia/assets/36925400/082d076d-a0af-45ae-a981-55128616a13b">

**Comment by gursahibsahni** [comment]
FWIW - I am using the following codec registry in my datastore config

`        final var pojoCodecRegistry = fromRegistries(MongoClientSettings.getDefaultCodecRegistry(),                fromProviders(PojoCodecProvider.builder().automatic(true).build()));`

**Comment by evanchooly** [comment]
If you could put together a [reproducer](/morphiaorg/reproducer) I'd be happy to take a look and see. This case is pretty well tested and years old so there must be something else in play I'm not seeing here.

**Comment by gursahibsahni** [comment]
@evanchooly I have added you to the project I am working on. I think that should help you to see the problem. Please let me know if by anyway I can help you more to look into it. Thank you.

**Comment by gursahibsahni** [comment]
@evanchooly I used to reproducer repo and figured out the problem. I still don't know what was the root issue. But here is the thing - 
If you see the screenshot of the above mongo compass - there is no type discriminator to tell mongo how to deserialize. I think I was using an old version of Mongo and after the change in the version, the change wasn't percolated in IntelliJ (I hate when it does that). Now, when I have changed the mongo version, I see the `_t` field has a value and can retrieve the value.

![image](https://github.com/MorphiaOrg/morphia/assets/36925400/46cbae10-509f-41dc-94bf-7f095a641a3c)

We can close the issue.
